crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +375,122 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    /// Returns the oldest settlement event for which the accociated auction is
+    /// not yet populated in the database.
+    pub async fn get_settlement_without_auction(
+        &self,
+    ) -> Result<Option<domain::settlement::Event>, error::SettlementEvent> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_settlement_without_auction"])
+            .start_timer();
+
+        let mut ex = self.postgres.pool.begin().await.map_err(DatabaseError)?;
+        let event = database::settlements::get_settlement_without_auction(&mut ex)
+            .await
+            .map_err(DatabaseError)?
+            .map(|event| {
+                let event = settlement::Event {
+                    inner: eth::Event {
+                        block: u64::try_from(event.block_number)
+                            .map_err(|_| error::SettlementEvent::InvalidEvent)?
```

Comment:
If the database is storing a block number that cannot fit into u64, I think this is a database error or at most a data consistency error. 

I don't think we should create a new `Event` specific error variant for this (as I'm pretty sure no callsite will handle this error type in a particular and we should stop simply adding a new variant for every possible error and instead spend a bit of time to think of what error taxonomy makes sense from the consumer side).

Reply:
refactored errors, similar errors merge to become DatabaseError